### ISSUE TITLE
Capital letters break tld recognition

### ISSUE THREAD

**Issue by gauda** [sporkmonger/addressable#227]
It seems capital letters break the recognition of the top level domain:
`Addressable::URI.parse("http://www.google.com").tld 
=> "com"
Addressable::URI.parse("http://www.google.Com").tld
=> ""
Addressable::URI.parse("http://www.google.COM").tld
=> ""`
I'd expect the case should not matter?

**Comment by sporkmonger** [comment]
As a workaround, try normalizing first.

**Comment by gauda** [comment]
I thought of something like that, too. Unfortunately, there may be cases where the path contains capital letters, too. So when you normalize https://en.wiktionary.ORG/wiki/MIT to https://en.wiktionary.org/wiki/mit you get a completely different result (compare the linked websites). I think the normalization has to be done internally in addressable only for the host.

**Comment by KitaitiMakoto** [comment]
@gauda 
`Addressable::URI#normalize` method keeps path's case:

```ruby
Addressable::URI.parse('https://en.wiktionary.ORG/wiki/MIT').normalize
=> #<Addressable::URI:0x3fdaa4f08598 URI:https://en.wiktionary.org/wiki/MIT>
```

**Comment by gauda** [comment]
@KitaitiMakoto Thanks for pointing me in the right direction!

**Comment by sporkmonger** [comment]
Leaving issue open until it's actually addressed.

**Comment by sporkmonger** [comment]
Also normalization can be done per-component if so desired.

**Comment by sporkmonger** [comment]
Fixed in f032606.
